### PR TITLE
Support specifying custom SSH public key.

### LINKED ISSUE #739: hostvars['127.0.0.1']['ansible_system'] may not exist
### Expected behavior:

Installation completed without errors.

### Actual Behavior:

Installation completed with minor errors:
```
TASK [streisand-gateway : Attempt to open the instructions on Linux (if applicable). Errors in this task are ignored because the `xdg-open` command is not always available.] ***
fatal: [x.x.x.x]: FAILED! => {"failed": true, "msg": "The conditional check 'hostvars['127.0.0.1']['ansible_system'] == \"Linux\"' failed. The error was: error while evaluating conditional (hostvars['127.0.0.1']['ansible_system'] == \"Linux\"): 'dict object' has no attribute 'ansible_system'\n\nThe error appears to have been in '/media/veracrypt1/streisand/playbooks/roles/streisand-gateway/tasks/main.yml': line 199, column 9, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n      - name: Attempt to open the instructions on Linux (if applicable). Errors in this task are ignored because the `xdg-open` command is not always available.\n        ^ here\n"}
```

### Steps to Reproduce:

Run `./streisand`.

### Additional Details:

Really minor issue. I guess it should be fixed by replacing of
```
playbooks/roles/streisand-gateway/tasks/main.yml:202
- when: hostvars['127.0.0.1']['ansible_system'] == "Linux"
+ when: hostvars['127.0.0.1']['ansible_system'] is defined and hostvars['127.0.0.1']['ansible_system'] == "Linux"
```
and
```
playbooks/roles/streisand-gateway/tasks/main.yml:207
- when: hostvars['127.0.0.1']['ansible_system'] == "Darwin"
+ when: hostvars['127.0.0.1']['ansible_system'] is defined and hostvars['127.0.0.1']['ansible_system'] == "Darwin"
```

### PR DESCRIPTION
**Pull request by cpu**
Prior to this PR Streisand universally assumed the user's SSH private key was `~/.ssh/id_rsa`. The public component was then used to provision cloud instances.

This PR adds a new `streisand_ssh_private_key` var to the `global_vars/default-site.yml` file. The site-specific override for these defaults, `~/.streisand/site.yml` is now customized by `playbooks/customize.yml` when the user specifies that they want to use a different SSH key. The public component is expected to exist at the same file path with a `.pub` suffix to keep things simple.

This PR proactively sets the `ansible_ssh_private_key_file` for the Streisand  SSH key specified. This ensures Ansible will load & use this private key to access the streisand-host bypassing the need to use a ssh-agent (though it will play nicely if you do).

The validation playbook is updated to check that the  `streisand_ssh_private_key` exists. The corresponding shell based checks for the SSH key were removed from the `./streisand` wrapper script since they are now replaced by these Ansible tasks. Similarly the existing server playbook
now performs its own quick test of a raw_cmd on the server instead of doing a SSH check in the `./streisand` wrapper. This means things fail a little bit later in the process if you can't actually SSH to the
existing server but it isn't *much* later in the process and its nice to `rm` some shell script code.

Two unrelated bugs were fixed with the existing server provisioning:  the genesis role was not being saved as a fact on localhost causing the diagnostics to print "unknown" for the genesis role used. Secondly, because no `gather_facts: yes` for the localhost was run for the existing server provision other diagnostic information was missing. This was also the root cause of Issue #739. 

Resolves #739, #300 

I have tested this PR with a  few providers (Linode, DigitalOcean, Existing Server, Localhost) using my own SSH key loaded in an ssh-agent ahead of time as well as with a brand new ed25519 keypair that used a passphrase. Both cases worked as expected.

### CODE REVIEW

File: playbooks/customize.yml [hunk 1]
```diff
@@ -4,6 +4,10 @@
   gather_facts: no
 
   vars_prompt:
+    - name: streisand_ssh_private_key
+      prompt: "Enter the path to your SSH private key? Press enter for default "
```
Comment (alimakki) [inline playbooks/customize.yml]: "Enter the path to your SSH private key, or press enter for default" reads a bit better in my opinion.  Either way the question mark can be dropped.

File: playbooks/existing-server.yml [hunk 2]
```diff
@@ -3,6 +3,30 @@
...
+  tasks:
+    - block:
+        - raw: whoami
+          args:
+            executable: /bin/bash
+          changed_when: False
+      rescue:
+        - fail:
+            msg: "Unable to SSH to existing streisand-host.\nEnsure private key corresponding to \"{{ streisand_ssh_key }}\" is loaded in your SSH key agent.\nTry using `ssh-keygen -i {{ streisand_ssh_key }} to generate your key if it does not exist\n"
+
```
Comment (alimakki) [inline playbooks/existing-server.yml]: Good hint 👍

File: playbooks/customize.yml (hunk 1 above)
Comment (cpu) [reply playbooks/customize.yml]: Thanks! fixed in https://github.com/jlund/streisand/pull/970/commits/ef529d198bcf6643a39f5b6bc445c21ada698f48

**Comment by cpu** [comment]
> Would this be simpler if we could guarantee Ansible runs under an ssh-agent, either an existing one or a freshly launched one? I ask because my ssh-agent talks to my Yubikey, so in some sense I don't have an id_rsa, only an id_rsa.pub.

It doesn't sound simpler to me :-) I'll give it some thought but I likely won't work on it myself if you're interested in trying as a follow-up. If so creating a second issue with some prelim details and assigning it to yourself would be great. Thanks!

**Pull request merged**